rustls-cert-gen/src/cert.rs

```diff
@@ -0,0 +1,433 @@
+use rcgen::{
+	BasicConstraints, Certificate, CertificateParams, DistinguishedName, DnType,
+	DnValue::PrintableString, ExtendedKeyUsagePurpose, IsCa, KeyUsagePurpose, SanType,
+};
+use std::{fs::File, io, path::Path};
+
+#[derive(Debug, Clone)]
+/// Pem serialized Certificate and Pem serialized corresponding private key
+pub struct PemCertifiedKey {
+	pub cert_pem: String,
+	pub private_key_pem: String,
+}
+
+impl PemCertifiedKey {
+	pub fn write(&self, dir: &Path, name: &str) -> Result<(), io::Error> {
+		use std::io::Write;
+		std::fs::create_dir_all(dir)?;
+
+		let key_path = dir.join(format!("{name}.key.pem"));
+		let mut key_out = File::create(key_path)?;
+		write!(key_out, "{}", &self.private_key_pem)?;
+
+		let cert_path = dir.join(format!("{name}.pem"));
+		let mut cert_out = File::create(cert_path)?;
+		write!(cert_out, "{}", &self.cert_pem)?;
+
+		Ok(())
+	}
+}
+
+/// Builder to configure TLS [CertificateParams] to be finalized
+/// into either a Ca or an End-Entity.
+#[derive(Default)]
+pub struct CertificateBuilder {
+	params: CertificateParams,
+}
+
+impl CertificateBuilder {
+	/// Initialize `CertificateParams` with defaults
+	/// # Example
+	/// ```
+	/// # use rustls_cert_gen::CertificateBuilder;
+	/// let cert = CertificateBuilder::new();
+	/// ```
+	pub fn new() -> Self {
+		let mut params = CertificateParams::default();
+		// override default Common Name
+		params.distinguished_name = DistinguishedName::new();
+		Self { params }
+	}
+	/// Set signature algorithm (instead of default). Returns `anyhow::Result<Self>`.
```

Comment:
Nit: I think this is duplication with the fn signature.
```suggestion
	/// Set signature algorithm (instead of default).
```